Foundation/UserDefaults.swift

```diff
@@ -73,147 +131,103 @@ open class UserDefaults: NSObject {
-        let cVal = bVal._swiftObject
-        enum convErr: Swift.Error {
-            case convErr
-        }
-        do {
-            let dVal = try cVal.map({ (key, val) -> (String, Any) in
-                if let strKey = key as? NSString {
-                    return (strKey._swiftObject, val)
-                } else {
-                    throw convErr.convErr
-                }
-            })
-            var eVal = [String : Any]()
-            
-            for (key, value) in dVal {
-                eVal[key] = value
-            }
-            
-            return eVal
-        } catch _ { }
-        return nil
+        return object(forKey: defaultName) as? [String: Any]
     }
+    
     open func data(forKey defaultName: String) -> Data? {
-        guard let aVal = object(forKey: defaultName),
-              let bVal = aVal as? NSData else {
-            return nil
-        }
-        return Data(referencing: bVal)
+        return object(forKey: defaultName) as? Data
     }
+    
     open func stringArray(forKey defaultName: String) -> [String]? {
-        guard let aVal = object(forKey: defaultName),
-              let bVal = aVal as? NSArray else {
-            return nil
-        }
-        return _SwiftValue.fetch(nonOptional: bVal) as? [String]
+        return object(forKey: defaultName) as? [String]
     }
+    
     open func integer(forKey defaultName: String) -> Int {
         guard let aVal = object(forKey: defaultName) else {
             return 0
         }
-        if let bVal = aVal as? NSNumber {
-            return bVal.intValue
+        if let bVal = aVal as? Int {
+            return bVal
         }
-        if let bVal = aVal as? NSString {
-            return bVal.integerValue
+        if let bVal = aVal as? String {
+            return NSString(string: bVal).integerValue
         }
         return 0
     }
+    
     open func float(forKey defaultName: String) -> Float {
         guard let aVal = object(forKey: defaultName) else {
             return 0
         }
-        if let bVal = aVal as? NSNumber {
-            return bVal.floatValue
+        if let bVal = aVal as? Float {
+            return bVal
         }
-        if let bVal = aVal as? NSString {
-            return bVal.floatValue
+        if let bVal = aVal as? String {
+            return NSString(string: bVal).floatValue
         }
         return 0
     }
+    
     open func double(forKey defaultName: String) -> Double {
         guard let aVal = object(forKey: defaultName) else {
             return 0
         }
-        if let bVal = aVal as? NSNumber {
-            return bVal.doubleValue
+        if let bVal = aVal as? Double {
+            return bVal
         }
-        if let bVal = aVal as? NSString {
-            return bVal.doubleValue
+        if let bVal = aVal as? String {
+            return NSString(string: bVal).doubleValue
         }
         return 0
     }
+    
     open func bool(forKey defaultName: String) -> Bool {
```

Comment:
One minor incompatibility between this and what darwin does, which we probably don’t want/need to fix: NSUserDefaults historically coerced values to int/bool/float using different logic from CF. I changed it so that subclasses maintain the old behavior (for compatibility reasons), but un-subclassed NSUserDefaults always uses the CF logic.

I haven’t checked whether this implements the NSUserDefaults version or the CFPrefs version, but I think we should use the CFPrefs behavior.